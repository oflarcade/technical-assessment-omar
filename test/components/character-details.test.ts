import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import CharacterDetails from '@/components/rick-and-morty/character-details.vue';
import CharacterImage from '@/components/rick-and-morty/character-image.vue';


const character = {
  id: 1,
  name: 'Rick Sanchez',
  status: 'Alive',
  species: 'Human',
  type: '',
  gender: 'Male',
  origin: {
    name: 'Earth',
    url: '',
  },
  location: {
    name: 'Earth',
    url: '',
  },
  image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
  episode: [],
  url: '',
  created: '2017-11-04T18:48:46.250Z',
};

describe('RickAndMortyCharacterDetails.vue', () => {
  it('renders character details correctly', () => {
    const wrapper = mount(CharacterDetails, {
      props: {
        character,
      },
      global: {
        components: {
          CharacterImage,
        },
      },
    });

    
    expect(wrapper.find('h2').text()).toBe('Rick Sanchez');
    expect(wrapper.findAll('p').at(0)?.text()).toBe('Name: Rick Sanchez');
    expect(wrapper.findAll('p').at(1)?.text()).toBe('Species: Human');
    expect(wrapper.findAll('p').at(2)?.text()).toBe('Status: Alive');
    expect(wrapper.findAll('p').at(3)?.text()).toBe('Type:');
    expect(wrapper.findAll('p').at(4)?.text()).toBe('Gender: Male');
    expect(wrapper.findAll('p').at(5)?.text()).toBe('Origin: Earth');
    expect(wrapper.findAll('p').at(6)?.text()).toBe('Location: Earth');
    expect(wrapper.findAll('p').at(7)?.text()).toContain('Created: November 4, 2017');

    
    const imageComponent = wrapper.findComponent(CharacterImage);
    expect(imageComponent.exists()).toBe(true);
    expect(imageComponent.props().character).toEqual(character);
  });

  it('matches snapshot', () => {
    const wrapper = mount(CharacterDetails, {
      props: {
        character,
      },
      global: {
        components: {
          CharacterImage,
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
