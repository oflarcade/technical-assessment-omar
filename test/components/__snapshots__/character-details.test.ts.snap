// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RickAndMortyCharacterDetails.vue > matches snapshot 1`] = `
"<div data-v-6f052df1="" class="border-4 border-pink-600 flex flex-col rounded-lg bg-slate-800">
  <header data-v-6f052df1="" class="p-2 flex items-center bg-pink-950 border-b border-b-4 border-pink-600">
    <h2 data-v-6f052df1="" class="font-bold text-lg capitalize text-pink-700">Rick Sanchez</h2>
  </header>
  <div data-v-6f052df1="" class="p-4">
    <p data-v-6f052df1="" class="text-sm">Name: Rick Sanchez</p>
    <p data-v-6f052df1="" class="text-sm">Species: Human</p>
    <p data-v-6f052df1="" class="text-sm">Status: Alive</p>
    <p data-v-6f052df1="" class="text-sm">Type: </p>
    <p data-v-6f052df1="" class="text-sm">Gender: Male</p>
    <p data-v-6f052df1="" class="text-sm">Origin: Earth</p>
    <p data-v-6f052df1="" class="text-sm">Location: Earth</p>
    <p data-v-6f052df1="" class="text-sm"> Created: November 4, 2017</p>
  </div>
  <div data-v-6f052df1="" class="h-[400px]"><img data-v-594c78aa="" data-v-6f052df1="" data-nuxt-img="" class="image w-full h-full object-cover rounded-lg" src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"></div>
</div>"
`;
